src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -185,15 +200,21 @@ def learning_type(self) -> LearningType:
 
     @property
     def transform(self) -> Transform:
-        """Retrieve the transform that the model should use during inference.
+        """Retrieve the model-specific transform.
 
-        If the model is attached to a trainer and the trainer has a datamodule with an eval_transform,
-        then the model will use that transform. Otherwise, it will use the default transform.
+        If a transform has been set using `set_transform`, it will be returned. Otherwise, we will use the
+        model-specific default transform, conditioned on the input size.
         """
-        if self._trainer and self._trainer.datamodule and self._trainer.datamodule.eval_transform:
-            return self._trainer.datamodule.eval_transform
+        if self._transform is not None:
+            return self._transform
+        if self._trainer and self.trainer.datamodule:
+            return self.configure_transforms(self.trainer.datamodule.image_size)
         return self.configure_transforms()
 
+    def set_transform(self, transform: Transform) -> None:
```

Comment:
What's the advantage of this over `@transform.setter`?

Reply:
setters don't play nice with Lightning for some reason. When we use a setter here, calling it will not properly set the attribute.

Reply:
I'd also prefer setter

Reply:
Using a setter is not possible here. Setting an attribute of a `LightningModule` using a setter does not work when the target attribute is a torch Module. It will just ignore the setter call. I'm not sure exactly why this is the case, but I guess it interferes with some internals of the `LightningModule`.